src/shared/utils.ts

```diff
@@ -1490,3 +1491,16 @@ export function newVote(voteType: VoteType, myVote?: number): number {
     return myVote == -1 ? 0 : -1;
   }
 }
+
+function sleep<T>(millis: number): Promise<T> {
+  return new Promise(resolve => setTimeout(resolve, millis));
+}
+
+/**
+ * Polls / repeatedly runs a promise, every X milliseconds
+ */
+export async function poll(promiseFn: any, millis: number) {
+  await promiseFn();
```

Comment:
In theory, this should be enough, but I'm not able to test it myself at the moment:

```suggestion
  if (window.document.visibilityState !== "hidden") {
    await promiseFn();
  }
```

Reply:
Ohh I see so this prevents it if you're looking at other tabs?

Reply:
Yup, exactly

Reply:
K sweet, I'll test that shortly.